draft-ietf-quic-tls.md

```diff
@@ -1269,20 +1269,12 @@ both protected with 1-RTT keys and contains an `ACK` frame.
 
 ### Updates to Data and Stream Limits
 
-`MAX_DATA`, `MAX_STREAM_DATA`, `BLOCKED`, `STREAM_BLOCKED`, and `MAX_STREAM_ID`
-frames MUST NOT be sent unprotected.
+`MAX_DATA`, `BLOCKED`, `MAX_STREAM_ID`, and `STREAM_ID_BLOCKED` frames MUST NOT
+be sent unprotected.
 
-Though data is exchanged on stream 0, the initial flow control window on that
-stream is sufficiently large to allow the TLS handshake to complete.  This
-limits the maximum size of the TLS handshake and would prevent a server or
-client from using an abnormally large certificate chain.
-
-Stream 0 is exempt from the connection-level flow control window.
-
-Consequently, there is no need to signal being blocked on flow control.
-
-Similarly, there is no need to increase the number of allowed streams until the
-handshake completes.
+`MAX_STREAM_DATA` and `STREAM_BLOCKED` frames MUST NOT be sent unprotected
+unless they identify stream 0.  Stream 0 is exempt from the connection-level
+flow control window and stream limits.
```

Comment:
But isn't the whole problem here that tampering with these packets can't be detected?  An attacker could cause one side to believe that it had a huge flow control window on stream 0, which could then cause the connection to blow up at some future point if they attempted to send to that limit.  Subsequent, lower MAX_STREAM_DATA frames would be ignored, because they don't increase the largest-seen value.

Reply:
Yeah, I'm prepared to acknowledge that as a limitation.  Perhaps we could say that the flow control limit is reset to the lower of the initial limit and the amount used once the handshake completes.

F**c I hate stream 0.

Reply:
Yeah, me too.  That flow control reset idea sounds unpleasant, but probably the best we can do if we have to tolerate window updates in unprotected packets.